weblab/entities/tests/test_views.py

```diff
@@ -652,6 +652,290 @@ def test_returns_404_if_commit_not_found(self, client, logged_in_user):
         assert response.status_code == 404
 
 
+@pytest.mark.django_db
+class TestEntityCompareFittingResultsView:
+    def test_shows_fittings_related_to_model_version(self, client, fittingresult_version):
+        fit = fittingresult_version.fittingresult
+
+        # should not be included, as it uses a different model and version
+        recipes.fittingresult_version.make()
+
+        # should not be included, as it uses a different version of this model
+        recipes.fittingresult_version.make(fittingresult__model=fit.model)
+
+        response = client.get(
+            '/entities/models/%d/versions/%s/fittings' % (fit.model.pk, fit.model_version.sha)
+        )
+
+        assert response.status_code == 200
+        assert response.context['comparisons'] == [(fit.dataset, [fit])]
+
+    def test_groups_by_dataset_for_model(self, client, public_model):
+        model_version = public_model.repocache.latest_version
+        dataset1, dataset2 = recipes.dataset.make(_quantity=2, visibility='public')
+
```

Comment:
Similarly to above, can we add some private versions and check the corresponding results aren't included?